agent.go

```diff
@@ -57,6 +59,9 @@ type sandbox struct {
 	grpcListener net.Listener
 	sharedPidNs  namespace
 	mounts       []string
+
+	procChLock  sync.RWMutex
+	processesCh map[int]chan int
```

Comment:
The code is not clean for me.

I hope the new exec_process.go is renamed to subreaper.go or reaper.go
And `type Subreaper struct {xx}` in the file.
Move the above two fields to the struct. sync.RWMutex can be anonymous for some convenience.
Move the related functions to be the functions of `type Subreaper`.
Other new functions in this PR may need to be moved to the file too.

Reply:
done